samples/csharp/getting-started/DatabaseIntegration/DatabaseIntegration/Program.cs

```diff
@@ -0,0 +1,122 @@
+                    });
+
+                // Add the data into the database
+                db.AdultCensus.AddRange(data);
+
+                var count = db.SaveChanges();
+                Console.WriteLine($"Save results {count}");
+            }
+        }
+
+        public static void Main()
+        {
+            // Seed the database with the dataset.
+            Seeding();
+
+            using(var db = new AdultCensusContext())
+            {
+                // Set the query tracking behavior to be NoTracking, this is particularly useful because
+                // our scenarios our read-only with the data. Not having tracking will  
+                db.ChangeTracker.QueryTrackingBehavior = QueryTrackingBehavior.NoTracking;
+                var mlContext =  new MLContext();
+               
+                // Query our training data from the database. This query is selecting everything from the AdultCensus table. The
+                // result is then loaded by ML.Net through the LoadFromEnumerable. LoadFromEnumerable returns an IDataView which
+                // can be consumed by an ML.Net pipeline.
+                // NOTE: For training, ML.Net requires that the training data is processed in the same order to produce consistent results.
+                // Therefore we are sorting the data by the AdultCensusId, which is an auto-generated id.
+                var dataView = mlContext.Data.LoadFromEnumerable(db.AdultCensus.OrderBy(x=>x.AdultCensusId).ToList<AdultCensus>());
+                var pipeline = mlContext.Transforms.Categorical.OneHotEncoding(new [] {
+                        new InputOutputColumnPair("MsOHE", "MaritalStatus"), 
+                        new InputOutputColumnPair("OccOHE", "Occupation"),
+                        new InputOutputColumnPair("RelOHE", "Relationship"),
+                        new InputOutputColumnPair("SOHE", "Sex"), 
+                        new InputOutputColumnPair("NatOHE", "NativeCountry")
+                    }, OneHotEncodingEstimator.OutputKind.Binary)
+                    .Append(mlContext.Transforms.Concatenate("Features", "MsOHE", "OccOHE", "RelOHE", "SOHE", "NatOHE"))
+                    .Append(mlContext.BinaryClassification.Trainers.LightGbm());
+
+                Console.WriteLine("Training model...");
+                var model = pipeline.Fit(dataView);
+
+                Console.WriteLine("Predicting...");
+                // Now that the model is trained, we want to test it's prediction results, which is done by using a test dataset
+                // and transforming it by the trained model.
+                // To retrieve our test data, this issues a second query, retrieving the first 2000 records, then converts to an IDataView
+                // in order to feed into the ML.Net pipeline.
+                var testDataView = mlContext.Data.LoadFromEnumerable(db.AdultCensus.Take(2000).ToList<AdultCensus>());
```

Comment:
Idem with the ToList here, althought it is less important because you have already limited the size of the sample with Take.